src/lib/err/torerr.c

```diff
@@ -118,8 +120,15 @@ tor_log_set_sigsafe_err_fds(const int *fds, int n)
     n = TOR_SIGSAFE_LOG_MAX_FDS;
   }
 
-  memcpy(sigsafe_log_fds, fds, n * sizeof(int));
-  n_sigsafe_log_fds = n;
+  /* Clear the entire array. There's a race condition here: err logs are
+   * disabled while the array is cleared. But that's better than accessing a
+   * partially written array. */
```

Comment:
I agree about the race condition -- but this change does not solve it completely.  Another thread can still read n_sigsafe_log_fds before it is changed, then read the new values of sigsafe_log_fds after they are zeroed, or while they are partially written.

To solve this right, we would probably need to do something with atomics, which we are not currently exposing at this level.

I do not think problem should block merge of this branch: it is still an improvement over the status quo.

Reply:
Ok, I will update the comment so that it's clear there's still some race here, and why.

Reply:
Fixed in 671ec51163ec69e8b632bb121ae60354b9bd5b1d